task1_materials_castelar/models/stock_move.py

```diff
@@ -0,0 +1,17 @@
+from odoo import api, models
+from odoo.exceptions import UserError
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+    
+    @api.onchange('product_uom_qty','quantity_done')
+    def _check_quantity_done(self):
+        '''
+        Checks if quantity_done must be equal or less than the demand for receipts in the Invetory
+        App
+        :param self - set of records associated to the context where this function is being called
+        '''
+        for record in self:
+            picking_type_record = self.env['stock.picking.type'].search([('id','=',record.picking_type_id.id)])
```

Comment:
You searched for a record you already have? record.picking_type_id == picking_type_record
Also if you were to have the id of a record, you use browse instead of search.